targets/REV3-Hardmon/main.cpp

```diff
@@ -151,13 +208,205 @@ int main() {
+                                                                             ACC_CAN_RECEIVE_THREAD_AUTOSTART);
+
+    rtos::Initializable* initArr[] = {
+        &hardmon, &modelThread, &modelTriggerFlag, &modelTriggerTimer,
+        &powertrainCANReceiveThread, &healthThread, &accessoryCanReceiveThread};
+
+    log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Starting Kernel.");
+
+    rtos::startKernel(initArr, sizeof(initArr) / sizeof(initArr[0]), txPool);
+}
+
+/**
+ * Triggers every time the model timer runs its course. Sets the first flag of the modelTriggerFlag to 1
+ *
+ * @param modelTriggerFlag the eventFlags that controls the model triggering.
+ */
+void modelTimerExpiration(rtos::EventFlags* modelTriggerFlag) {
+    uint32_t flags;
+    modelTriggerFlag->getCurrentFlags(&flags);
+    if ((flags & 0x01) == 0x1) {
+        //THE MODEL IS NOT RUNNING FAST ENOUGH THROW AN ERROR
+        //todo: determine what error to throw
+        log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Model Thread not running fast enough.");
+    }
+    modelTriggerFlag->set(0x01);
+}
+
+/**
+ * Entry Function for the thread that runs the model. Waits for the modelTimer to
+ * set the modelTriggerFlag to run one step of the process method
+ *
+ * @param args the arguments for this thread
+ */
+[[noreturn]] void modelThreadEntry(modelThreadArgs_t* args) {
+    rtos::TXError error;
+    log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Model Thread Started.");
+    while (true) {
+        uint32_t flagOutput;
+        args->triggerFlag->get(0x01, true, true, rtos::TXWait::TXW_WAIT_FOREVER, &flagOutput);
+        log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Model Starting.");
+        args->hardmon->process();
+        log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Model Processed.");
+    }
+}
+
+/**
+ * Entry Function for the powertrainCANThread.
+ *
+ * @param args the arguments for this thread
+ */
+[[noreturn]] void powertrainCANReceiveThreadEntry(powertrainCANReceiveThreadArgs_t* args) {
+    log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Powertrain CAN Thread Started.");
+    io::CANMessage message;
+    rtos::Queue* queue = args->hardmon->getPowertrainQueue();
+    while (true) {
+        //suspends if there are no messages to receive
+        queue->receive(&message, rtos::TXWait::TXW_WAIT_FOREVER);
+        args->hardmon->handlePowertrainCanMessage(message);
+    }
+}
+
+/**
+ * Entry Function for the healthThread.
+ *
+ * @param args the arguments for this thread
+ */
+[[noreturn]] void healthThreadEntry(healthThreadArgs_t* args) {
+    log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Health Thread Started.");
+
+    rtos::TXError error;
+    while (true) {
+        log::LOGGER.log(core::log::Logger::LogLevel::DEBUG, "Health Thread Triggered.");
+        //do healththread stuff
```

Comment:
I'm just nitpicking at this point ngl 💀.
```suggestion
        //do healthThread stuff
```